cmd/gateway/main.go

```diff
@@ -4,10 +4,212 @@
 package main
 
 import (
-	"storj.io/storj/cmd/uplink/cmd"
+	"context"
+	"crypto/rand"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+
+	base58 "github.com/jbenet/go-base58"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap"
+
+	"storj.io/storj/internal/fpath"
+	"storj.io/storj/pkg/cfgstruct"
+	"storj.io/storj/pkg/miniogw"
 	"storj.io/storj/pkg/process"
+	"storj.io/storj/pkg/storage/streams"
+	"storj.io/storj/pkg/storj"
+)
+
+// GatewayFlags configuration flags
+type GatewayFlags struct {
+	APIKey             string `default:"" help:"the api key to use for the satellite" setup:"true"`
+	GenerateMinioCerts bool   `default:"false" help:"generate sample TLS certs for Minio GW" setup:"true"`
+	SatelliteAddr      string `default:"localhost:7778" help:"the address to use for the satellite" setup:"true"`
+
+	miniogw.Config
+}
+
+var (
+	// GWCmd represents the base gateway command when called without any subcommands
+	GWCmd = &cobra.Command{
+		Use:   "gateway",
+		Short: "The Storj client-side S3 gateway",
+	}
+	setupCmd = &cobra.Command{
+		Use:         "setup",
+		Short:       "Create an gateway config file",
+		RunE:        cmdSetup,
+		Annotations: map[string]string{"type": "setup"},
+	}
+	runCmd = &cobra.Command{
+		Use:   "run",
+		Short: "Run the S3 gateway",
+		RunE:  cmdRun,
+	}
+
+	defaultConfDir     = fpath.ApplicationDir("storj", "gateway")
+	defaultIdentityDir = fpath.ApplicationDir("storj", "identity", "gateway")
```

Comment:
Same here, I think it might have used uplink folders so that it's easier setup... but it can be left separate, either is fine.